### PR TITLE
Add analytics 1

### PR DESCRIPTION
**Pull request by Aichnerc**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently there is no Google Analytics and/or Facebook analytics and/or Hotjar.
- Currently the code quality is very bad.
- Currently there is no cookie modal.

**What is the new behavior (if this is a feature change)?**
- Google Analytics, Facebook and Hotjar has been implemented.
- The code quality has been moved forwards by 2000 years.
- A cookie modal has been added.

**Other information**:
- May the Dark Side guide your path.

### CODE REVIEW

File: src/components/molecules/CookieModal/index.jsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+  save() {
+    localStorage.setItem(
+      "cookie",
+      JSON.stringify({
+        essential: this.state.essential ? true : false,
+        statistics: this.state.statistics ? true : false,
+        marketing: this.state.marketing ? true : false,
+      })
+    );
```
Comment (schettn) [inline src/components/molecules/CookieModal/index.jsx:48]: add newline

File: src/components/molecules/Footer/index.jsx [hunk 2]
```diff
@@ -6,111 +6,112 @@ import { Link } from "react-router-dom";
...
+              <p>
+                SithCult is a worldwide Star Wars fan organization comprised of
+                and operated by Star Wars fans.
+              </p>
+            </MDBCol>
+            <MDBCol md="3">
+              <h5 className="title">Public Services</h5>
+              <ul>
+                <Link to="/basic">
+                  <li className="list-unstyled">Lern Imperial Basic</li>
```
Comment (schettn) [inline src/components/molecules/Footer/index.jsx:31]: What is meant by "Lern"?

File: src/components/molecules/Footer/index.jsx [hunk 3]
```diff
@@ -6,111 +6,112 @@ import { Link } from "react-router-dom";
...
+                    />
+                  </li>
+                </a>
+                <a
+                  href="https://www.patreon.com/sithcult"
+                  target="_blank"
+                  rel="noopener noreferrer"
+                >
+                  <li className="list-unstyled">
+                    Support of on Patreon
```
Comment (schettn) [inline src/components/molecules/Footer/index.jsx:72]: Do you mean "support us"?

File: src/components/molecules/Footer/index.jsx [hunk 4]
```diff
@@ -6,111 +6,112 @@ import { Link } from "react-router-dom";
...
           <MDBContainer>
-          <p>SithCult is a worldwide Star Wars fan-club organization comprised of 
-          and operated by Star Wars fans. While it is not sponsored by Lucasfilm Ltd., 
-          it it follows generally accepted ground rules for Star Wars fan groups. Star 
-          Wars, its characters, costumes, and all associated items are the intellectual 
-          property of Lucasfilm. © & ™ Lucasfilm Ltd. All rights reserved.</p>
+            <p>
+              SithCult is a worldwide Star Wars fan-club organization comprised
+              of and operated by Star Wars fans. While it is not sponsored by
+              Lucasfilm Ltd., it it follows generally accepted ground rules for
```
Comment (schettn) [inline src/components/molecules/Footer/index.jsx:110]: Duplicate "it"

File: src/components/molecules/Navbar/index.jsx [hunk 5]
```diff
@@ -60,96 +59,102 @@ class Navbar extends React.Component{
 
     console.log(profile);
```
Comment (schettn) [inline src/components/molecules/Navbar/index.jsx:60]: Is this still necessary?

File: src/components/molecules/modals/AboutUs/index.jsx [hunk 6]
```diff
@@ -1,93 +1,100 @@
...
   state = {
-    modal11: false
-  }
+    modal11: false,
+  };
 
-  toggle = nr => () => {
-    let modalNumber = 'modal' + nr
+  toggle = (nr) => () => {
+    let modalNumber = "modal" + nr;
```
Comment (schettn) [inline src/components/molecules/modals/AboutUs/index.jsx:25]: add newline

File: src/components/pages/LoginPage/index.jsx [hunk 7]
```diff
@@ -24,126 +24,127 @@ import {
...
   render() {
     const { authErrorDetails, auth, location } = this.props;
 
-    let params = location.search.substr(1) ? location.search.substr(1).split("=") : null;
-    if(params){
-      if(params[0] === "refer"){
-        switch(params[1]){
+    let params = location.search.substr(1)
+      ? location.search.substr(1).split("=")
+      : null;
```
Comment (schettn) [inline src/components/pages/LoginPage/index.jsx:77]: add newline

File: src/components/pages/MessagePage/index.jsx [hunk 8]
```diff
@@ -21,7 +21,7 @@ import "./messagepage.scss";
 class MessagePage extends React.Component {
   render() {
     const { auth, profile } = this.props;
-    console.log(auth,profile);
+    console.log(auth, profile);
```
Comment (schettn) [inline src/components/pages/MessagePage/index.jsx:24]: Is this still necessary?

File: src/config/fbInit.js [hunk 9]
```diff
@@ -1,25 +1,25 @@
...
+import "firebase/firestore";
+import "firebase/auth";
+import "firebase/analytics";
 
 // Import secret credentials
-import config from './fbConfig.js'
+import config from "./fbConfig.js";
 
 // Init Firebase
 firebase.initializeApp(config);
```
Comment (schettn) [inline src/config/fbInit.js:11]: add newline

File: src/store/actions/authActions.js [hunk 10]
```diff
@@ -1,128 +1,143 @@
...
 export const signUp = (newUser) => {
-    return (dispatch, getState, {getFirebase, getFirestore}) => {
-        const firebase = getFirebase();
-        const firestore = getFirestore();
+  return (dispatch, getState, { getFirebase, getFirestore }) => {
+    const firebase = getFirebase();
+    const firestore = getFirestore();
 
-        console.log(newUser);
+    console.log(newUser);
```
Comment (schettn) [inline src/store/actions/authActions.js:42]: Is this still necessary?

File: src/store/actions/authActions.js [hunk 11]
```diff
@@ -1,128 +1,143 @@
...
-            if(!duplicate){
-                // Set default values
-                let credits, reputation, title, badges, status, department, beta, skin;
+    // Check for Sithname duplicates
+    firestore
+      .collection("users")
+      .get()
+      .then((querySnapshot) => {
+        let result = undefined;
+        let duplicate = false;
```
Comment (schettn) [inline src/store/actions/authActions.js:50]: add newline

File: src/store/actions/authActions.js [hunk 12]
```diff
@@ -1,128 +1,143 @@
...
+        querySnapshot.forEach(function (doc) {
+          // doc.data() is never undefined for query doc snapshots
+          result = doc.data();
```
Comment (schettn) [inline src/store/actions/authActions.js:53]: add newline

File: src/store/actions/postActions.js [hunk 13]
```diff
@@ -1,154 +1,188 @@
...
-        }, { merge: true }).then(() => {
-            dispatch({ type: 'LIKE_SUCCESS', id });
-            return 
-        }).catch((err) => {
-            dispatch({ type: 'LIKE_ERROR', err });
-        })
+  return (dispatch, getState, { getFirebase, getFirestore }) => {
+    const firebase = getFirebase();
+    const firestore = getFirestore();
+
```
Comment (schettn) [inline src/store/actions/postActions.js:35]: Remove newline

File: src/store/actions/postActions.js [hunk 14]
```diff
@@ -1,154 +1,188 @@
...
+    let localLikes = likes;
```
Comment (schettn) [inline src/store/actions/postActions.js:36]: add newline

File: src/store/actions/postActions.js [hunk 15]
```diff
@@ -1,154 +1,188 @@
...
-        } else {
-            dispatch({ type: 'REMOVE_ERROR', err: "Not authorized." });
-        }
+export const loadPosts = (amount) => {
+  return (dispatch, getState, { getFirebase, getFirestore }) => {
+    const firebase = getFirebase();
+    const firestore = getFirestore();
+    dispatch({ type: "LOAD_LOADING" });
+
+    let posts = firestore.collection("posts").where("visible", "==", true);
```
Comment (schettn) [inline src/store/actions/postActions.js:131]:
```suggestion
    let posts = firestore.collection("posts").where("visible", "==", true);
    
    dispatch({ type: "LOAD_LOADING" });

```
Is that possible? If so, adjust this. If not check this area for newlines.

File: src/store/actions/postActions.js [hunk 16]
```diff
@@ -1,154 +1,188 @@
...
-            dispatch({ type: 'LOAD_SUCCESS', results });
-        })
-        .catch((err) => {
-            dispatch({ type: 'LOAD_ERROR', err });
+    posts
+      .orderBy("timestamp", "desc")
+      .limit(amount)
+      .get()
+      .then((querySnapshot) => {
+        let results = [];
```
Comment (schettn) [inline src/store/actions/postActions.js:142]: add newline

File: src/store/actions/postActions.js [hunk 17]
```diff
@@ -1,154 +1,188 @@
...
+        querySnapshot.forEach(function (doc) {
+          let data = doc.data();
```
Comment (schettn) [inline src/store/actions/postActions.js:144]: add newline

File: src/store/actions/postActions.js [hunk 18]
```diff
@@ -1,154 +1,188 @@
...
-        })
-        .catch((err) => {
-            dispatch({ type: 'LOAD_ERROR', err });
-        });
+  return (dispatch, getState, { getFirebase, getFirestore }) => {
+    const firebase = getFirebase();
+    const firestore = getFirestore();
+    dispatch({ type: "LOAD_LOADING" });
+
+    let posts = firestore.collection("posts");
```
Comment (schettn) [inline src/store/actions/postActions.js:161]:
```suggestion
    let posts = firestore.collection("posts");
 
    dispatch({ type: "LOAD_LOADING" });

   
```
Is that possible? If so, adjust this. If not check this area for newlines.

File: src/store/actions/postActions.js [hunk 19]
```diff
@@ -1,154 +1,188 @@
...
     }
-}
 
-/** 
+    posts
+      .orderBy("timestamp", "desc")
+      .limit(amount)
+      .get()
+      .then((querySnapshot) => {
+        let results = [];
```
Comment (schettn) [inline src/store/actions/postActions.js:172]: add newline

File: src/store/actions/postActions.js [hunk 20]
```diff
@@ -1,154 +1,188 @@
...
+        querySnapshot.forEach(function (doc) {
+          let data = doc.data();
```
Comment (schettn) [inline src/store/actions/postActions.js:174]: add newline

File: src/store/actions/postActions.js [hunk 21]
```diff
@@ -1,154 +1,188 @@
...
+          results.push({ id: doc.id, data });
+        });
```
Comment (schettn) [inline src/store/actions/postActions.js:176]: add newline

File: src/store/actions/postActions.js [hunk 22]
```diff
@@ -1,154 +1,188 @@
...
+    posts
+      .orderBy("timestamp", "desc")
+      .limit(amount)
+      .get()
+      .then((querySnapshot) => {
+        let results = [];
+        querySnapshot.forEach(function (doc) {
+          let data = doc.data();
+          results.push({ id: doc.id, data });
         });
```
Comment (schettn) [inline src/store/actions/postActions.js:146]: add newline

File: src/store/actions/userActions.js [hunk 23]
```diff
@@ -1,37 +1,40 @@
...
+    const firebase = getFirebase();
+    const firestore = getFirestore();
 
-        // Get loading
-        dispatch({ type: 'USERLOAD_LOADING' });
+    // Get loading
+    dispatch({ type: "USERLOAD_LOADING" });
 
-        console.log(uid);
+    console.log(uid);
```
Comment (schettn) [inline src/store/actions/userActions.js:9]: Is this still neccesary?

File: src/store/actions/userActions.js [hunk 24]
```diff
@@ -1,37 +1,40 @@
...
-        });
-    }
-}
+    firestore
+      .collection("users")
+      .doc(uid)
+      .get()
+      .then((doc) => {
+        if (!doc.exists) {
+          console.log("User no longer exists!");
```
Comment (schettn) [inline src/store/actions/userActions.js:17]: Is this still necessary?

File: src/store/actions/userActions.js [hunk 25]
```diff
@@ -1,37 +1,40 @@
...
+          dispatch({ type: "USERLOAD_ERROR", err: "User no longer exists." });
+        } else {
+          console.log("Document data:", doc.data());
```
Comment (schettn) [inline src/store/actions/userActions.js:20]: Is this still necessary?

File: src/store/actions/postActions.js (hunk 15 above)
Comment (Aichnerc) [reply src/store/actions/postActions.js:131]: Nope

File: src/components/organisms/RegisterForm/index.jsx [hunk 26]
```diff
@@ -315,79 +312,86 @@ class HomePage extends React.Component {
...
       "human",
       "ogre",
-      "orc"
+      "orc",
     ];
 
     /**
      * Genders
      * This is not meant ot offend persons with different genders. The module used
      * requires clear inputs acknowledging only female and male as valid parameters.
```
Comment (kleberbaum) [inline src/components/organisms/RegisterForm/index.jsx:321]: Uff, got already a little bit triggered. xD Y is ur module acting like a privileged white male tho? LMAO

**Comment by kleberbaum** [comment]
You should consider adding codacy to this repo.

File: src/components/organisms/RegisterForm/index.jsx (hunk 26 above)
Comment (Aichnerc) [reply src/components/organisms/RegisterForm/index.jsx:321]: 👀

**Pull request merged**